components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -73,65 +87,166 @@ public AbstractHTTPFunction() {
         allowedDomains = ConfigProvider.getInstance().getAllowedDomainsForHttpFunctions();
     }
 
-    protected void executeHttpMethod(HttpUriRequest request, Map<String, Object> eventHandlers) {
+    protected void executeHttpMethod(HttpUriRequest clientRequest, Map<String, Object> eventHandlers,
+                                     AuthConfigModel authConfigModel) {
 
         AsyncProcess asyncProcess = new AsyncProcess((context, asyncReturn) -> {
-            JSONObject json = null;
-            int responseCode;
             String outcome;
             String endpointURL = null;
 
-            if (request.getURI() != null) {
-                endpointURL = request.getURI().toString();
-            }
+            HttpUriRequest request;
+            try {
+                if (authConfigModel != null) {
+                    AuthConfig authConfig = AuthConfigFactory.getAuthConfig(authConfigModel, context, asyncReturn);
+                    request = authConfig.applyAuth(clientRequest, authConfigModel);
+                } else {
+                    request = clientRequest;
+                }
 
-            if (!isValidRequestDomain(request.getURI())) {
-                outcome = Constants.OUTCOME_FAIL;
-                LOG.error("Provided Url does not contain a allowed domain. Invalid Url: " + endpointURL);
-                asyncReturn.accept(context, Collections.emptyMap(), outcome);
-                return;
-            }
+                if (request.getURI() != null) {
+                    endpointURL = request.getURI().toString();
+                }
 
-            try (CloseableHttpResponse response = client.execute(request)) {
-                responseCode = response.getStatusLine().getStatusCode();
-                if (responseCode >= 200 && responseCode < 300) {
-                    outcome = Constants.OUTCOME_SUCCESS;
-                    if (response.getEntity() != null) {
-                        Header contentType = response.getEntity().getContentType();
-                        String jsonString = EntityUtils.toString(response.getEntity());
-                        if (contentType != null && contentType.getValue().contains(TYPE_TEXT_PLAIN)) {
-                            // For 'text/plain', put the response body into the JSON object as a single field.
-                            json = new JSONObject();
-                            json.put(RESPONSE, jsonString);
-                        } else {
-                            JSONParser parser = new JSONParser();
-                            json = (JSONObject) parser.parse(jsonString);
+                if (!isValidRequestDomain(request.getURI())) {
+                    LOG.error("Provided Url does not contain a allowed domain. Invalid Url: " + endpointURL);
```

Comment:
Can we simplify this as "Request URL does not match with the allowed domain list. Allowed domains: {domain list} Request URL: {request uri}"

Reply:
Resolved with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/files#diff-0e83b3b35fef53c4408b95506e56fdd6bac553873dba30f0e8e8d646f1223313R413

Reply:
Noted that we have changed a debug log to an error log, which is not a good move.
My suggestion was just to simplify the message.
Given that the debug is added to reveal allowed domain lists and mismatch we can just say "Request URL does not match with the allowed domain list. Request Url: {request uri}"

Using the word "contain" does not read right